crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -0,0 +1,257 @@
+    web3::signing::SecretKeyRef,
+};
+
+#[tokio::test]
+#[ignore]
+async fn price_improvement_fee_sell_order() {
+    run_test(price_improvement_fee_sell_order_test).await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn price_improvement_fee_sell_order_capped() {
+    run_test(price_improvement_fee_sell_order_capped_test).await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn price_improvement_fee_buy_order() {
+    run_test(price_improvement_fee_buy_order_test).await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn price_improvement_fee_buy_order_capped() {
+    run_test(price_improvement_fee_buy_order_capped_test).await;
+}
+
+// Without protocol fee:
+// Expected execution is 10000000000000000000 GNO for
+// 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
+//
+// With protocol fee:
+// surplus in buy token = 9871415430342266811 - 5000000000000000000 =
+// 4871415430342266811
+//
+// protocol fee in buy token = 0.3*surplus = 1461424629102680043
+//
+// protocol fee in sell token = 1461424629102680043 / 9871415430342266811 *
+// (10000000000000000000 - 167058994203399) = 1480436341679873337
+//
+// expected executed_surplus_fee is 167058994203399 + 1480436341679873337 =
+// 1480603400674076736
+async fn price_improvement_fee_sell_order_test(web3: Web3) {
+    execute_test(
+        web3.clone(),
+        "--fee-policy-kind=priceImprovement:0.3:1.0".to_string(),
+        OrderKind::Sell,
+        1480603400674076736u128.into(),
+    )
+    .await;
+}
+
+// Without protocol fee:
+// Expected executed_surplus_fee is 167058994203399
+//
+// With protocol fee:
+// Expected executed_surplus_fee is 167058994203399 +
+// 0.1*10000000000000000000 = 1000167058994203400
+async fn price_improvement_fee_sell_order_capped_test(web3: Web3) {
+    execute_test(
+        web3.clone(),
+        "--fee-policy-kind=priceImprovement:1.0:0.1".to_string(),
+        OrderKind::Sell,
+        1000167058994203400u128.into(),
+    )
+    .await;
+}
+
+// Without protocol fee:
+// Expected execution is 5040413426236634210 GNO for 5000000000000000000 DAI,
+// with executed_surplus_fee = 167058994203399 GNO
+//
+// With protocol fee:
+// surplus in sell token = 10000000000000000000 - 5040413426236634210 =
+// 4959586573763365790
+//
+// protocol fee in sell token = 0.3*4959586573763365790 = 1487875972129009737
+//
+// expected executed_surplus_fee is 167058994203399 + 1487875972129009737 =
+// 1488043031123213136
+async fn price_improvement_fee_buy_order_test(web3: Web3) {
+    execute_test(
+        web3.clone(),
+        "--fee-policy-kind=priceImprovement:0.3:1.0".to_string(),
+        OrderKind::Buy,
+        1488043031123213136u128.into(),
+    )
+    .await;
+}
+
+// Without protocol fee:
+// Expected execution is 5040413426236634210 GNO for 5000000000000000000 DAI,
+// with executed_surplus_fee = 167058994203399 GNO
+//
+// With protocol fee:
+// Expected executed_surplus_fee is 167058994203399 + 0.1*5040413426236634210 =
+// 504208401617866820
+
```

Comment:
Comments are much more readable with spaces

Reply:
I don't want you to get rid of all the line breaks. 😅  Just this particular empty line because it separates the comment from the test function. All other comments are directly attached to the test they are talking about.

Reply:
gotcha! 😄 fixed